MyOnboarding/test/MyOnboardingApp.Services.Tests/Services/CreateItemServiceTests.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MyOnboardingApp.Contracts.Errors;
+using MyOnboardingApp.Contracts.Generators;
+using MyOnboardingApp.Contracts.Models;
+using MyOnboardingApp.Contracts.Repository;
+using MyOnboardingApp.Contracts.Validation;
+using MyOnboardingApp.Services.Services;
+using MyOnboardingApp.TestUtils.Extensions;
+using MyOnboardingApp.TestUtils.Factories;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace MyOnboardingApp.Services.Tests.Services
+{
+    [TestFixture]
+    public class CreateItemServiceTests
+    {
+        [Test]
+        public async Task AddNewItemAsync_ValidItemSpecified_ReturnsAddedItemWithNoError()
+        {
+            var repository = Substitute.For<ITodoListRepository>();
+            var idGenerator = Substitute.For<IIdGenerator<Guid>>();
+            var dateTimeGenerator = Substitute.For<IDateTimeGenerator>();
+            var validator = Substitute.For<IInvariantValidator<TodoListItem>>();
+            var service = new CreateItemService(repository, idGenerator, dateTimeGenerator, validator);
+
+            var itemToAdd = new TodoListItem
+            {
+                Id = Guid.Empty,
+                Text = "Item to Test",
+                CreationTime = DateTime.MinValue,
+                LastUpdateTime = DateTime.MinValue
+            };
+            var expectedId = new Guid("00000000-0000-0000-0000-000000000007");
+            idGenerator
+                .GetNewId()
+                .Returns(expectedId);
+            var expectedDateTime = new DateTime(2018, 10, 10);
+            dateTimeGenerator
+                .GetCurrentDateTime()
+                .Returns(expectedDateTime);
+            var (_, _, expectedResult) = ItemVariantsFactory
+                .CreateItemVariants(
+                    expectedId,
+                    itemToAdd.Text,
+                    expectedDateTime);
+            validator
+                .Validate(Arg.Any<TodoListItem>())
+                .Returns(callInfo 
+                    => ItemWithErrors.Create(callInfo.Arg<TodoListItem>(), new List<Error>().AsReadOnly()));
```

Comment:
```suggestion
                    => ItemWithErrors.Create(callInfo.Arg<TodoListItem>(), Enumerable.Empty<Error>()));
```
(based on https://github.com/petrareb/kentico-onboarding-cs/pull/4#discussion_r263424985)